_posts/2017-08-08-CompetitiveCoding.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: post
+title: "Competitive Coding: From Novice to Expert"
+date: 2017-08-08
+author: Pushkar Dixit
+
+
+tags: 
+ - Competitive Programming
+ - Introduction
+categories:
+ - Competitive programming
+ 
+image:
+  url: https://superiorstudents.files.wordpress.com/2017/05/book-wordle.png
+---
+ 
+# Learning the way to learn
+
+
+<center>
+  <img src="https://superiorstudents.files.wordpress.com/2017/05/book-wordle.png" style="width: 35%; margin-left: auto; margin-right: auto;">
+</center>
+ 
+Though this post doesn't really presents an introduction to fundamentals of competitive programming but highlights some extremely important points which can really mark the difference between a novice coder and a veteran. Sometimes learning the way to learn can be important, so these points can help you become a better learner at least because competitive coding is all about learning, learning and learning.
+
+
+### Choose a suitable language and know your language well enough
+ 
+Though there are plenty of options to choose from when it comes to choosing a language for development tasks, the truth is there are not many when you got to choose one for competitive programming. It's either C/C++ or JAVA. That's it. You need to go with one or the other. More than three-fourths of the coders use C/C++ so is advisable to go with it and try to use other languages python/JAVA only when a specific question demands it. Like 'Very Easy' type problems can be quickly coded in python so many experienced coders resort to python to make a quick submission for such problems in short contests. Never stick to a particular language like JAVA just for the sake you have been doing all your stuff in it from your school days and now are hating to switch to a different language. JAVA will certainly give you a hard time coding solutions because it's syntax is lengthier and that is enough to set your mood off. 
+But JAVA too has lots of things to boast off as well. Even though it is slower, Java has powerful built-in libraries and APIs such as BigInteger/BigDecimal, GregorianCalendar, Regex, etc.Java programs are easier to debug with the virtual machine’s ability to provide a stack trace when it crashes (as opposed to core dumps or segmentation faults in C/C++).
```

Comment:
GregorianCalendar, Regex have no use in Competitive Programming. Also you can track C/C++ if you know gdb :)